components/driver/sdmmc_host.c

```diff
@@ -68,16 +68,22 @@ static void configure_pin_iomux(uint8_t gpio_num);
 
 static esp_err_t sdmmc_host_pullup_en_internal(int slot, int width);
 
-void sdmmc_host_reset(void)
+esp_err_t sdmmc_host_reset(void)
 {
     // Set reset bits
     SDMMC.ctrl.controller_reset = 1;
     SDMMC.ctrl.dma_reset = 1;
     SDMMC.ctrl.fifo_reset = 1;
+
     // Wait for the reset bits to be cleared by hardware
+    uint32_t t0 = esp_log_timestamp();
     while (SDMMC.ctrl.controller_reset || SDMMC.ctrl.fifo_reset || SDMMC.ctrl.dma_reset) {
-        ;
+        if (esp_log_timestamp() - t0 > 1000) {
```

Comment:
perhaps use `ESP_RETURN_ON_FALSE` (from esp_check.h) for this sort of check?

Reply:
Additionally, I wouldn't rely on esp_log_timestamp, it is not guaranteed to be monotonic. I would suggest esp_timer_get_time, instead.

Reply:
I looked at `ESP_RETURN_ON_FALSE`, but I noticed it logs, which we don't need here.

Reply:
@chipweinberger Fair point on logging, I'm on the fence if the timeout should be logged or not. It would be really nice if `ESP_RETURN_ON_FALSE` had a way to pass in a log level or a flag to say "don't log this"